aws_lambda_powertools/logging/logger.py

```diff
@@ -1180,6 +1194,23 @@ def flush_buffer(self) -> None:
         Any exceptions from underlying logging or buffer mechanisms
         will be propagated to caller
         """
+        # Check ALC level against buffer level
+        lambda_log_level = self._get_aws_lambda_log_level()
+        if lambda_log_level:
+            # Check if buffer level is less verbose than ALC
+            if (
+                hasattr(self, "_buffer_config")
+                and self._buffer_config
```

Comment:
If you move this function after the check of empty buffer then you can remove this 2 conditions and optimize the code/cpu cycle. Pls try that.